### ISSUE TITLE
Feature Request: CloudTrail Organizations Trail Support

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#6579]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS recently released support for CloudTrail trails to span across entire AWS Organizations: https://aws.amazon.com/about-aws/whats-new/2018/11/aws-cloudtrail-adds-support-for-aws-organizations/

> You can now create, manage, and deploy AWS CloudTrail trails across an organization from a single account. Through integration with AWS Organizations, this organization trail enables you to define a uniform event logging strategy for your organization that is applied automatically to each member account in an organization. Users in member accounts are able to see these trails, but they can’t modify them. This helps you uniformly apply and enforce your event logging strategy across the AWS accounts in your organization.

> Using your organization's master account, you can set up a trail and mark it for deployment across the member accounts. Then, AWS CloudTrail automatically copies the trail definition to each member account upon creation and propagates changes to the member accounts when the trail in the master account is updated. Organization trails capture events for all accounts in the organization and log them to the same Amazon S3 bucket as defined by the organization trail in the master account.

### Affected Resource(s)

* `aws_cloudtrail`

### Potential Terraform Configuration

```hcl
# Not currently implemented, details may change during development
resource "aws_cloudtrail" "example" {
  is_organization_trail = true
}
```

### References

* https://aws.amazon.com/about-aws/whats-new/2018/11/aws-cloudtrail-adds-support-for-aws-organizations/
* https://github.com/aws/aws-sdk-go/blob/master/CHANGELOG.md#release-v11579-2018-11-20
* https://docs.aws.amazon.com/awscloudtrail/latest/APIReference/API_CreateTrail.html

**Comment by gwvandesteeg** [comment]
Exactly what I was after for creating org wide cloudtrails

**Comment by bflad** [comment]
The implementation is relatively straightforward -- its just the acceptance testing that is proving slightly difficult since we need to create an organization and appropriately enable CloudTrail service access to the organization. Here's the initial pull request, but the testing still needs some adjustments to support Organization service access: #6580

**Comment by gwvandesteeg** [comment]
Tested using the created branch, this worked perfectly for creating Organisation wide CloudTrails, thanks.

**Comment by bflad** [comment]
Support for this has been merged in and will release with version 1.51.0 of the AWS provider, likely middle of next week. 👍
